src/app/loadout-builder/generated-sets/GeneratedSet.tsx

```diff
@@ -50,18 +52,35 @@ function GeneratedSet({
     return null;
   }
 
-  const assignedMods = $featureFlags.armor2ModPicker
+  const [assignedMods] = $featureFlags.armor2ModPicker
     ? assignModsToArmorSet(
         set.armor.map((items) => items[0]),
         lockedArmor2Mods
       )
-    : {};
+    : [{}];
+
+  const canCompareLoadouts =
+    set.armor.every((items) => items[0].classType === selectedStore?.classType) &&
+    loadouts.some((l) => l.classType === selectedStore?.classType);
 
   return (
     <div className={styles.build} style={style} ref={forwardedRef}>
       <div className={styles.header}>
-        <SetStats defs={defs} set={set} statOrder={statOrder} enabledStats={enabledStats} />
-        <GeneratedSetButtons set={set} store={selectedStore!} onLoadoutSet={setCreateLoadout} />
+        <SetStats
+          defs={defs}
+          stats={set.stats}
+          items={set.armor.map((items) => items[0])}
```

Comment:
Not sure it matters here, but this'll force a re-render every time. Not a big deal if every time this renders, it needs to render.

Reply:
It might trigger on a stores refresh but in a usual workflow it will only render if they change the set, and then it needs to render again anyways.